### DISCUSSION TITLE
The order of panel widgets in the system section.

### DISCUSSION THREAD

**Discussion by korenevskiy** [joomla/joomla-cms#35019] [2. Cold Features]
Colleagues, by what principle are the link panels in the system section sorted?
I would prefer the SETUP and MANAGE panels to be the first.
I feel tension every time I open a panel and try to find the right panel somewhere in the middle of other panels. And if the screen is small, then you will have to search for the most popular panel by rewinding the page.
.
Let's place the SETUP and MANAGE panels first.

**Comment by brianteeman** [comment]
There is a module position at the top called icon. Below that is a module position called cpanel. 
The modules are displayed (within each position) sorted by ordering and then displayed left to right.

**Comment by korenevskiy** [comment]
@brianteeman 
I put it wrong. I meant this.
![image](https://user-images.githubusercontent.com/6898474/127853303-ed6658ab-beb2-4d56-8bab-84feb7ac9c46.png)
Who determined this order of panels? By what principle is this order formed?
It's hard for me to search for the MANAGE panel every time. It can be in different places, but it should always be at the beginning.
I marked with numbers how I would like to see the order. But, I may be inaccurate. The main thing is that I would like to see the SETUP and MANAGE the in  first position.

**Reply by brianteeman** [reply]
Honestly dont remember how the ordering was done. However its is done someone will prefer something else. The file they are defined in is here https://github.com/joomla/joomla-cms/blob/4.0-dev/administrator/components/com_menus/presets/system.xml

You can always create your own admin modules and publish them here instead or as well

**Reply by korenevskiy** [reply]
Help, I changed the order in the presets. The order has not changed. I have been looking for a long time for the code and searching for files, according to the contents, I can't find where the placement takes place. Or rather, I can't find where the GRID row templates are defined.

**Comment by korenevskiy** [comment]
@RickR2H  @dgrammatiko  @rdeutz  @brianteeman @richard67 
I used the old float: left property;
That's cool! The layout is stable, the tiles do NOT jump.
It looks exactly like a "Masonry".
It loads quickly, accurately and is pleasant to the eyes.
.
"Masonry" markup has not yet been introduced into CSS. And for these purposes, we are introducing JS "Masonry". Colleagues is this really justified?
.
Let's also think about something else, when the community implements "Masonry" in CSS, it will be completely different and it will have different property settings. In any case, we will have to redo the layout of the settings.
.
I will load the layout from FLOAT:LEFT.
And you will test and compare. You will see that this is much better than "Masonry".
Metaphor: In our city there are motorists who remake a clearance of 2-3 cm in semi-rotten(old) cars. They think they're cool. It looks very strange.
It seems to me that our "Masonry" is the same.
We played around and tested it, we realized that the idea did not fit. We will return to it when "Masonry" appears in CSS.

I will create a PR. And you will test and compare.
.
Tell me how to disable JS "Masonry" here?

**Reply by RickR2H** [reply]
I agree that the UX could be better. But it works for now and I don't think it is dealbreaker for the first stable release. Never the less we have to rethink the implementation.

**Reply by korenevskiy** [reply]
> I agree that the UX could be better. But it works for now and I don't think it is dealbreaker for the first stable release. Never the less we have to rethink the implementation. 

if you would have seen the version with float: left in the case, then you would have understood that such an amendment is undoubtedly worthy of a stable version.
Because jumping when loading is a piece of trash. sorry for the comparison
And if we rethink the UX, it will certainly not be included in the stable version. The new UX requires the approval of the majority and long improvements in any case.
I am ready to take part in the rethinking of UX. This is already the work of Joom4. 1

**Comment by korenevskiy** [comment]
> No need to vote thats not how we do things - things will be even slower to develop

The Masonry layout implies automatic placement depending on the height of the block. So that the lower border of the parent's space is aligned with the content.
Thus, the order of the blocks was imposed on us by the JS Masonry script. At your discretion. But in the cards there were blocks in which the number of elements is the same. Even they can't be swapped at our discretion. The JS script has its own logic.
I turned off Masonry. Sorted the blocks as the creator/maintainer of the sites.
As a result, it was possible to remove the block jumps when the page was loading.
Please test the PR

https://github.com/joomla/joomla-cms/pull/35040
